fog/ledger/enclave/api/src/messages.rs

```diff
@@ -1,19 +1,45 @@
 // Copyright (c) 2018-2021 The MobileCoin Foundation
 
 //! The message types used by the ledger_enclave_api.
-
-use fog_types::ledger::{CheckKeyImagesResponse, GetOutputsResponse};
+use crate::UntrustedKeyImageQueryResponse;
+use alloc::vec::Vec;
+use fog_types::ledger::GetOutputsResponse;
 use mc_attest_core::{Quote, Report, TargetInfo, VerificationReport};
 use mc_attest_enclave_api::{ClientAuthRequest, ClientSession, EnclaveMessage};
 use mc_common::ResponderId;
+use mc_transaction_core::ring_signature::KeyImage;
 use serde::{Deserialize, Serialize};
 
+/// A struct representing the key image stores data
+#[derive(
+    Serialize,
+    Deserialize,
+    Debug,
+    Clone,
+    Copy,
+    PartialEq,
+    PartialOrd,
+    core::cmp::Eq,
+    core::hash::Hash,
+    Ord,
+)]
+pub struct KeyImageData {
```

Comment:
I think the code comments on this struct can be improved:
when we document data in a struct, we usually want to focus on like "what is the meaning of this piece of data, how can i know if this field is being set and used correctly", and like, what are the relationships between these data.

the comments right now say things like "We use in the ORAM here and here", but what I'd like it to say is something like this:

```
pub struct KeyImageData {
    /// A key image which has appeared in the blockchain
    pub key_image: KeyImage,
    /// The index of the block in which this key image appeared
    pub block_index: u64,
    /// The timestamp of the block in which this key image appeared
    pub timestamp: u64,
}
```

This is much more useful because if someone thinks there is a problem with block index, they can check what the `block_index` field is documented to be, and then check if someone is setting it to a value that's different from the documentation, or using it with the idea that it means something else.